src/components/charts/ReferencesSearch.tsx

```diff
@@ -0,0 +1,86 @@
+import _ from 'lodash'
+import React from 'react'
+import { Search } from 'semantic-ui-react'
+import { AirtableRecord } from '../../types';
+import Translate from "../../utils/translate/translateService";
+                                                        
+const initialState = {
+  loading: false,
+  results: [],
+  value: '',
+}
+
+const MAX_SUGGESTIONS = 5
+
+function searchReducer(state : any, action : any) {
+  switch (action.type) {
+    case 'CLEAN_QUERY':
+      return initialState
+    case 'START_SEARCH':
+      return { ...state, loading: true, value: action.query }
+    case 'FINISH_SEARCH':
+      return { ...state, loading: false, results: action.results }
+    case 'UPDATE_SELECTION':
+      return { ...state, value: action.selection }
+
+    default:
+      throw new Error()
+  }
+}
+
+const resultRenderer = ({ source_name, publish_date } : any) => {
+  return (<React.Fragment>
+            <div>{source_name}</div>
+            <span className="text-muted">{Translate('Published')}: {publish_date}</span>
+          </React.Fragment>)
+}
+
+function ReferenceSearch(props : any) {
+  const { source, onSearchResultChange } = props;
+  const [state, dispatch] = React.useReducer(searchReducer, initialState)
+  const { loading, results, value } = state
+
+  const timeoutRef : any = React.useRef()
+
+  const handleSearchChange = React.useCallback((e, data) => {
+    clearTimeout(timeoutRef.current)
+    dispatch({ type: 'START_SEARCH', query: data.value })
+
+    timeoutRef.current = setTimeout(() => {
```

Comment:
why do we need a timeout here? can searches really take that long?

Reply:
This timeout is used for executing the search, so after 800 milliseconds since the users last keystroke, the search will be executed. I'm just following semantic-ui's design for this one